maintainers/maintainer-list.nix

```diff
@@ -15939,4 +15939,13 @@
     github = "wuyoli";
     githubId = 104238274;
   };
+  DomesticMoth = {
```

Comment:
please keep the file in alphabetical order and put this in a separate commit: `maintainers: add DomesticMoth`